app/src/main/java/com/sun/dogbreeds/ui/search/SearchViewModel.kt

```diff
@@ -1,9 +1,48 @@
 package com.sun.dogbreeds.ui.search
 
+import androidx.lifecycle.MutableLiveData
+import com.sun.dogbreeds.coroutine.CoroutineResult
+import com.sun.dogbreeds.data.db.entity.Breed
+import com.sun.dogbreeds.data.repsitory.BreedsLocalRepository
 import com.sun.dogbreeds.ui.base.BaseViewModel
+import com.sun.dogbreeds.utils.match
+import kotlinx.coroutines.launch
 
-class SearchViewModel : BaseViewModel(){
+class SearchViewModel(private val repository: BreedsLocalRepository) : BaseViewModel() {
+
+    val recyclerViewData = MutableLiveData<List<Breed>>()
```

Comment:
```suggestion
    val recyclerViewData = MutableLiveData<List<Breed>>()
```
nên private nhưng fileds mutable và có thể expose immutable 
```
private val _breeds = MutableLiveData<List<Breed>>()
val breeds : LiveData<List<Breed>>
        get() = _breeds
```